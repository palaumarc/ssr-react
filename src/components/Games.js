import React from "react";
import { connect } from "react-redux";
import { Link } from "react-router-dom";
import { fetchGames } from "../actions";
import { getGames } from "../reducers";

class Games extends React.Component {

    render( ) {
        const { games } = this.props;

        return (
            <div>
                <h2>Games</h2>
                <ul>
                    { games.map( ( { id, names, assets } ) => (
                        <Link key={id} to={`/games/${id}`} >
                            { names.international } - <img src={assets["cover-tiny"].uri} />
                        </Link>
                    ) ) }
                </ul>
            </div>
        );
    }
}

Games.serverFetch = fetchGames; // static declaration of data requirements

const mapStateToProps = (state) => ({
    games: getGames(state)
});

export default connect(mapStateToProps)(Games);
