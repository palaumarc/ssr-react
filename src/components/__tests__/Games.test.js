import React from 'react';
import Games from '../Games';
import renderer from 'react-test-renderer';

jest.mock('../GamePreview', () => 'GamePreview');

it('Game props is an array of 2 valid objects. Snapshot', () => {
    const game1 = {
        id: 1,
        name: 'dummyName1',
        logoUrl: 'dummyLogoUrl1'
    }

    const game2 = {
        id: 2,
        name: 'dummyName2',
        logoUrl: 'dummyLogoUrl2'
    }

    const tree = renderer.create(<Games games={[game1, game2]} />).toJSON();
    expect(tree).toMatchSnapshot();
});