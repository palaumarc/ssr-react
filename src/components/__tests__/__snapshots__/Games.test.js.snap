// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game props is an array of 2 valid objects. Snapshot 1`] = `
<div>
  <h2>
    Games
  </h2>
  <div
    style={
      Object {
        "display": "grid",
        "gridGap": "5px",
        "gridTemplateColumns": "repeat(auto-fill, minmax(250px, 1fr))",
      }
    }
  >
    <link-mock
      to="/games/1"
    >
      <GamePreview
        logoUrl="dummyLogoUrl1"
        name="dummyName1"
      />
    </link-mock>
    <link-mock
      to="/games/2"
    >
      <GamePreview
        logoUrl="dummyLogoUrl2"
        name="dummyName2"
      />
    </link-mock>
  </div>
</div>
`;
